### ISSUE TITLE
endingDay & startingDay are not displayed correct on Android.

### ISSUE THREAD

**Issue by seco35** [wix/react-native-calendars#642]
## Description

endingDay and startingDay =true doesn't works properly on Android working without Expo. It works when using expo.

## Expected Behavior

Marking with rounded edges.

## Observed Behavior

Marking as a rectangle ( as if it had startingDay & endingDay = false).

<img width="368" alt="bildschirmfoto 2018-09-24 um 12 35 17" src="https://user-images.githubusercontent.com/25101244/45947936-598be300-bff6-11e8-9f84-6d6eb91e690f.png">

## Environment

Please run these commands in the project folder and fill in their results:

* `npm ls react-native-calendars`: 1.20.0
* `npm ls react-native`: 0.57.1

Emulator Nexus 5 : API 28 (Android 9)
Huawei Mediapad M5: API 26 (Android 8)

## Reproducible Demo

mDates["2018-10-05"] = { startingDay: true, color: lockColor, textColor:'black', endingDay: true, selection:false}

**Comment by adimshev** [comment]
+1

**Comment by adimshev** [comment]
![image](https://user-images.githubusercontent.com/33316685/48701366-61839000-ebff-11e8-8762-5f0cf5f80215.png)

**Comment by wojciechver** [comment]
+1

npm ls react-native-calendars: 1.21.0
npm ls react-native: 0.57.5

Having the same issue on android (works fine on iOS). 
It's only present after a selection is made on the calendar. If a period is sent to the calendar on open, it rounds the corners properly.

on load:
<img width="358" alt="screen shot 2018-11-23 at 14 53 34" src="https://user-images.githubusercontent.com/32451034/48946943-b260f480-ef2f-11e8-95fd-40cd587e8e53.png">


after new selection:
<img width="370" alt="screen shot 2018-11-23 at 14 51 09" src="https://user-images.githubusercontent.com/32451034/48946934-a6753280-ef2f-11e8-83c6-a888a5da83cb.png">

**Comment by adimshev** [comment]
You can use prop dayComponent to pass your own day component. It works for me.

**Comment by kalraneeraj24550** [comment]
plz help me, calnder doesnt give rounded period corner.
![screenshot from 2018-12-18 13-29-39](https://user-images.githubusercontent.com/44640902/50139891-ffcf4800-02c8-11e9-8c3d-f30ee0ba1b12.png)
getDateString(timestamp) {
        const date = new Date(timestamp)
        const year = date.getFullYear()
        const month = date.getMonth() + 1
        const day = date.getDate()

        let dateString = `${year}-`
        if (month < 10) {
            dateString += `0${month}-`
        } else {
            dateString += `${month}-`
        }
        if (day < 10) {
            dateString += `0${day}`
        } else {
            dateString += day
        }

        return dateString;
    }

    getPeriod(startTimestamp, endTimestamp) {
        const period = {}
        let currentTimestamp = startTimestamp;
        while (currentTimestamp < endTimestamp) {
            const dateString = this.getDateString(currentTimestamp);
            period[dateString] = {
                color: '#0071f8',
                borderRadius: 100,
                startingDay: currentTimestamp === startTimestamp
            }
            currentTimestamp += 24 * 60 * 60 * 1000;
        }
        const dateString = this.getDateString(endTimestamp);
        period[dateString] = {
            color: '#0071f8',
             endingDay: true,
        }
        return period;
    }



    setDay(dayObj) {

        const _selectedDay = moment(dayObj.dateString).format(_format);
        console.log(dayObj.dateString)

        count++

            if (count === 1) {
                this.setState({
                    firstDay: dayObj.day,
                    firstMonth: dayObj.month,
                    firstYear: dayObj.year
                })
            }

            else if (count === 2) {
                this.setState({
                    secondday: dayObj.day,
                    secondMonth: dayObj.month,
                    secondYear: dayObj.year
                })
                count = 0
            }
        this.setState({
            todayDate: dayObj.dateString
        })
        const { start, end } = this.state;
        const {
            dateString, day, month, year,
        } = dayObj;
        let selected = true;
        const timestamp = new Date(year, month - 1, day).getTime()
        const newDayObj = { ...dayObj, timestamp }
   //     console.log('timestamp=', newDayObj);

        const startIsEmpty = _isEmpty(start);
     

        const _selectedDay1 = moment(day.dateString).format(_format);
        if (startIsEmpty || !startIsEmpty && !_isEmpty(end)) {
            const period = {
                [dateString]: {
                    color: '#0071f8',
                    endingDay: true,
                    startingDay: true,
                    selected: true,
                    
                   
                },

                ...{ [_selectedDay1]: { selected } }

            }
            this.setState({ start: newDayObj, period, end: {} })
        } else {
            // if end date is older than start date switch
            const { timestamp: savedTimestamp } = start
            if (savedTimestamp > timestamp) {
                const period = this.getPeriod(timestamp, savedTimestamp);
                this.setState({ start: newDayObj, end: start, period });
            } else {
                const period = this.getPeriod(savedTimestamp, timestamp);
                this.setState({ end: newDayObj, start, period });
            }
        }
    }




<Calendar
                        onDayPress={this.setDay.bind(this)}
                        markingType='period'
                        markedDates={period}
                        minDate={this.state.todayDate}
                        theme={{
                            calendarBackground: eachViewBackgroundColor,
                            textSectionTitleColor: '#D0D7DF',
                            dayTextColor: clickableDateTextColor,
                            selectedDayRadius: 100,
                            textDisabledColor: 'grey',
                            todayTextColor: '#543BEA',
                            monthTextColor: clickableDateTextColor,
                            arrowColor: '#0071f8',
                            selectedDayBackgroundColor: 'red',
                            selectedDayTextColor: 'white',

                        }}

/>

**Comment by afresh** [comment]
@kalraneeraj24550 https://github.com/afresh/react-native-calendars/commit/37b713c3922ae3282701cd1436ab9a478d9c2bed

**Comment by ahanriat** [comment]
Looks like a missing `overflow: 'hidden'` style on the **containerStyle** here https://github.com/wix/react-native-calendars/blob/master/src/calendar/day/period/index.js#L131

**Comment by linnett** [comment]
Does anyone know when this fix will go in?

**Comment by elgalesmana** [comment]
as mentioned from @ahanriat , to fix this we need add `overflow: hidden` into base stylesheet from day period. And since the PR still not merged https://github.com/wix/react-native-calendars/pull/728 , you can try to solve the issue like this :

```js
theme={{
  'stylesheet.day.period': {
      base: {
        overflow: 'hidden',
        height: 34,
        alignItems: 'center',
        width: 38,
      }
  }
}}
```

**Comment by tamer-mohamed** [comment]
@elgalesmana I tried your solution it works perfectly when the calendar is loaded with the marked dates.

<img width="384" alt="Screenshot 2019-04-29 at 09 17 42" src="https://user-images.githubusercontent.com/4436327/56881170-e51f9380-6a5f-11e9-88a1-5fd3a56b2364.png">


However, when I change the marked dates selection, it ends up with this result.

<img width="385" alt="Screenshot 2019-04-29 at 09 15 56" src="https://user-images.githubusercontent.com/4436327/56881173-e94bb100-6a5f-11e9-8a26-fd23d8fbf23d.png">


anything I'm doing wrong?

**Comment by elgalesmana** [comment]
@tamer-mohamed you need to update the `markedDates` state values. you can do this with `onDayPress` like in this plugin [https://github.com/lazaronixon/react-native-date-range-picker/blob/master/DateRangePicker.js](https://github.com/lazaronixon/react-native-date-range-picker/blob/master/DateRangePicker.js)

**Comment by tamer-mohamed** [comment]
@elgalesmana 
In the first load, marked dates are shown correctly.
I'm passing the `markedDates` from the state, but the broderRadius disspear from the starting date once the component is updated (once `markedDates` are changed)



```jsx

<Calendar 
  markedDates={this.props.markedDates}
  onDayPress={this._handleDayPress}
  theme={{'stylesheet.day.period': {
            base: {
              overflow: 'hidden',
              height: 34,
              alignItems: 'center',
              width: 38,
            }
   }}
/>


```

**Comment by buddhiv** [comment]
> @elgalesmana
> In the first load, marked dates are shown correctly.
> I'm passing the `markedDates` from the state, but the broderRadius disspear from the starting date once the component is updated (once `markedDates` are changed)
> 
> ```js-jsx
> <Calendar 
>   markedDates={this.props.markedDates}
>   onDayPress={this._handleDayPress}
>   theme={{'stylesheet.day.period': {
>             base: {
>               overflow: 'hidden',
>               height: 34,
>               alignItems: 'center',
>               width: 38,
>             }
>    }}
> />
> ```

Correct answer until the PR is merged, I experienced the same issue in 1.135.0. Thanks a lot
